### ISSUE TITLE
Preloaded Interstitial presentation takes too long

### ISSUE THREAD

**Issue by mrbodich** [AppLovin/AppLovin-MAX-Swift-Package#10]
It takes around 350 ms between `maInterstitialAd.show()` and `didDisplay(_ ad:)`, and same visually. Even if I pass ViewController explicitly to `show()`. Some short 50ms takes to present the Ad on the black screen also.
I experience same behavior in both your Demo App and in my app.
For the instance, AdMob Interstitial presented immediately when I call `gadInterstitialAd.present(fromRootViewController: vc)`, literally immediately, not just faster.

Is it possible to fix this please?

(see timestamps here)
<img width="337" alt="Screenshot 2024-07-30 at 5 59 51 PM" src="https://github.com/user-attachments/assets/d288d78a-94b5-49a7-b73e-42b38532b99f">

**Comment by thomasmso** [comment]
@mrbodich - the delay is intentional for various reasons and not configurable. What is the use case or issue that you are facing?

**Comment by mrbodich** [comment]
Hi @thomasmso 
The case: I need to show the Interstitial right when the user change the tab, or open another page BEFORE the user can see the page contents.

Can I kindly ask you to list the reasons of such delay?
Google for example, strictly disallows such behavior, you can read details in their docs by this link
https://support.google.com/admob/answer/6201362?sjid=4022034452363209116-NA#zippy=%2Cdisallowed-example-interstitial-launches-after-page-load
![CAD1F8260F59329A91195F4D196DD152BD88](https://github.com/user-attachments/assets/01998d62-da78-4a5a-8b63-103c0a6b845e)


Here is the piece of text from there:
> A common issue is that even though you may intend for the ad to load in between page content, the ad itself appears shortly after a new page of content has loaded due to carrier latency. To prevent this from happening, we recommend you pre-load the interstitial in advance.

Will appreciate if you will describe reasons why we must have a delay for interstitials, and can't just show the pre-loaded ad, because it's totally opposite to what Google stated and what they consider as an issue.

Thank you.
Regards.

**Comment by thomasmso** [comment]
Your use case is fair. In the next SDK release, we'll have a way for you to disable this 0.2s delay before an ad is shown.

**Comment by mrbodich** [comment]
@thomasmso Thank you!
Can you please mirror this update to Android SDK also? We use both, and Android team is experiencing the same delay.

**Comment by thomasmso** [comment]
@mrbodich - Sure thing, I'll keep this thread posted. Next SDK release is not scheduled yet, but I anticipate 2-3 weeks.

**Comment by thomasmso** [comment]
@mrbodich - we just released iOS SDK 12.6.1. You can disable the delay by setting an extra parameter as follows. Feel free to lmk if any Qs.

```
let sdk = ALSdk.shared()
sdk.settings.setExtraParameterForKey("fullscreen_ad_display_delay_enabled", value: "false")
sdk.initialize(with: configuration) { configuration in
...
```

**Comment by mrbodich** [comment]
Thank you @thomasmso! And yes we are waiting for the same feature for Android.
Thank you for pointing that out @andrey-kotkovets

**Comment by santoshbagadi** [comment]
@andrey-kotkovets @mrbodich yes, the new param was added to Android too and is available on Android SDK version 12.6.1 as well.
